### PR TITLE
Remove six in Cython code

### PR DESCRIPTION
**Pull request by okuta**
Remove `six` in Cython code for performance up.

### CODE REVIEW

**Comment by gwtnb** [comment]
Please resolve conflicts.

**Comment by okuta** [comment]
@gwtnb I resolved conflicts.

**Comment by gwtnb** [comment]
LGTM

File: cupy/cuda/pinned_memory.pyx [hunk 1]
```diff
@@ -321,6 +319,6 @@ cdef class PinnedMemoryPool:
             int: The total number of free blocks.
         """
         cdef Py_ssize_t n = 0
-        for v in six.itervalues(self._free):
+        for v in self._free.itervalues():
```
Comment (gwtnb) [inline cupy/cuda/pinned_memory.pyx]: Python 3 does not have `defaultdict.itervalues`.

Comment (okuta) [inline cupy/cuda/pinned_memory.pyx]: Yes, but Cython has it.

Comment (gwtnb) [inline cupy/cuda/pinned_memory.pyx]: I see.

Comment (sonots) [inline cupy/cuda/pinned_memory.pyx]: Let me take a note here. Actually, we got an error for `self._free.itervalues()` with python3. Fixed via https://github.com/cupy/cupy/pull/295

**Pull request merged**